### ISSUE TITLE
Tests depend on mirage but it's optional

### ISSUE THREAD

**Issue by dsheets** [mirage/ocaml-dns#76]
This cryptic error:

```
$ make
ocamlopt.opt -o setup.bin setup.ml || ocamlopt -o setup.bin setup.ml || ocamlc -o setup.bin setup.ml
rm -f setup.cmx setup.cmi setup.o setup.cmo
./setup.bin -configure --enable-lwt   --enable-tests  --prefix /usr/local
E: Cannot find buildable internal library 'dns-lwt-mirage' when checking build depends
E: Failure("1 configuration error")
Makefile:16: recipe for target 'setup.data' failed
make: *** [setup.data] Error 1
```

is probably because

```
Executable test
  Path:               lib_test/ounit
  MainIs:             test.ml
  Build$:             flag(tests)
  Custom:             true
  CompiledObject:     best
  Install:            false
  BuildDepends:       dns, dns.lwt-core, dns.mirage, io-page.unix, unix, oUnit, pcap-format, lwt.unix
```

but the `mirage` flag is false.

**Comment by infidel** [comment]
Fixed as part of PR #75, I think

**Comment by dsheets** [comment]
Ah! Thanks for that! It looks like you fixed it in 21bd7d2. I've restarted the jobs that failed due to nocrypto constraint. Will merge soon. Sorry for the delay.
